src/clerk.rs

```diff
@@ -0,0 +1,145 @@
+use std::time::{Duration, SystemTime, UNIX_EPOCH};
+
+use crate::config::Config;
+use clerk_rs::{
+    apis::{sessions_api, users_api},
+    clerk::Clerk,
+    models::{self, UpdateUserMetadataRequest},
+    ClerkConfiguration,
+};
+use secrecy::ExposeSecret;
+use serde_json::json;
+
+#[derive(Debug)]
+pub enum ClerkError {
+    InvalidSession,
+    FailedToGetUser,
+    FailedToUpdateUser,
+    Other(String), // For other errors, if needed
+}
+
+impl std::fmt::Display for ClerkError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            ClerkError::InvalidSession => write!(f, "Invalid session"),
+            ClerkError::FailedToGetUser => write!(f, "Failed to get user"),
+            ClerkError::FailedToUpdateUser => write!(f, "Failed to update user"),
+            ClerkError::Other(msg) => write!(f, "{}", msg),
+        }
+    }
+}
+
+impl std::error::Error for ClerkError {}
+
+pub struct ClerkResponse {
+    pub user: models::User,
+    pub session: models::Session,
+}
+
+pub struct ClerkHandler {
+    pub client: Clerk,
+    pub dispense_limit_interval: u64,
+}
+
+impl ClerkHandler {
+    // Initialize a new ClerkClientHandler with Clerk configuration
+    pub fn new(config: &Config) -> Self {
```

Comment:
Why do we want to create a client for each connection? Can we create it once at the start of the server and use it everywhere?